### PR TITLE
Rename config template to policy template and dataset to datastream

### PR DESCRIPTION
**Pull request by mtojek**
This PR renames config templates to policy templates and datasets to datastreams.

I regenerated golden files.

### CODE REVIEW

File: util/data_stream.go [hunk 1]
```diff
@@ -33,8 +33,8 @@ var validTypes = map[string]string{
 	"metrics": "Metrics",
 }
 
-type Dataset struct {
-	// Name and type of the dataset. This is linked to dataset.name and dataset.type fields.
+type DataStream struct {
+	// Name and type of the dataStream. This is linked to dataStream.name and dataStream.type fields.
 	Type string `config:"type" json:"type" validate:"required"`
 	Name string `config:"name" json:"name,omitempty" yaml:"name,omitempty"`
```
Comment (jen-huang) [inline util/data_stream.go:39]: `name` should be `dataset` following https://github.com/elastic/package-spec/pull/24/files#diff-13d2900dcce8e312a06c834b665816a1R71

Comment (mtojek) [reply util/data_stream.go:39]: Thanks for finding it! I'm looking into this.

Comment (mtojek) [reply util/data_stream.go]: Fixed

**Comment by ycombinator** [comment]
Nit: s/dataset/data stream/ in: https://github.com/elastic/package-registry/blob/868db0707fbc309fbd24c210259153800b47b93f/testdata/package/datasources/1.0.0/dataset/examplemetric/manifest.yml#L1

**Review by ycombinator — APPROVED** [review]
LGTM.

**Comment by mtojek** [comment]
> testdata/package/datasources/1.0.0/dataset/examplemetric/manifest.yml

Thanks, fixed!

**Pull request merged**